tensorflow/compiler/xla/service/gpu/launch_dimensions.cc

```diff
@@ -100,12 +99,20 @@ StatusOr<LaunchDimensions> CalculateLaunchDimensions(
   // TODO(jlebar): Investigate this further, and tune this heuristic so we can
   // run faster on the few benchmarks where smaller block size helps.
   int64 threads_per_block = ThreadsPerBlockLimit(gpu_device_info);
+  int64 threads_per_block_y = 1;
   int64 threads_per_block_row_vectorized =
       ThreadsPerBlockRowVectorized(shape, gpu_device_info, dim_config);
   if (threads_per_block_row_vectorized > 0) {
+    CHECK(dim_config.row_vectorized);
     threads_per_block = threads_per_block_row_vectorized;
-    VLOG(2) << "Update # of threads per block to (" << threads_per_block
+    if (threads_per_block < 128 && num_elements > 128) {
+      // This case happens for small row size.
+      threads_per_block_y =
+          CeilOfRatio((int64)128, threads_per_block);
```

Comment:
Nitpick: C-style casts are discouraged, maybe `128ll`? does it not work with just `128`?

Reply:
`128ll` doesn't work. `128l` works. But this looks brittle to me. On different platform "long int" have different size. So I guess it could break compilation on some platform.

The only good fix I think of would be to create more version of CeilOfRatio()
Currently it is templated to be CeilOfRatio(T, T)
We could create template CeilOfRatio(int, T) and CeilOfRatio(T, int) to prevent this.
But then we need to make sure we support all the variation including signed and unsigned version. Not something trivial to do without errors. Or maybe we can use advanced template to restrict T to be signed. I never did that before.

It is sad that C++ doesn't upcast with template. Here the upcast would be safe to do...
If you know a better fix, tell me. XLA has this problem at multiple places.

Reply:
Well actually in this case since `128` is used many times in the block, it's probably better to extract it to a constant with a meaningful name anyway.